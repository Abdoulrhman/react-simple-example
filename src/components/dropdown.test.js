import React from 'react'
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'

import Dropdown from './dropdown'

describe('Dropdown component', () => {
  let options;
  let props;
  let component;
  const label = 'models';
  Enzyme.configure({
    adapter: new Adapter()
  });

  beforeEach(() => {
    options = [
      { value: 'jeep', label: 'Jeep'},
      { value: 'tesla', label: 'Tesla'}
    ];
    props = {
      onChange: jest.fn()
    }

    component = shallow(
      <Dropdown 
        onChange={props.onChange} 
        label={label} 
        placeholder="- All Models -"  
        options={options} 
        selectedOption={options[0]} />
    )
  })

  it('should render the right way', () => {
    const component = shallow(<Dropdown />);
    expect(component).toMatchSnapshot()
  })

  it('should render label correctly', () => {
    expect(component).toMatchSnapshot()
  });

  it('should render label correctly', () => {
    expect(component).toMatchSnapshot()
    expect(component.find('label').text()).toBe(label+':');
  });

  it('should render options correctly', () => {
    expect(component).toMatchSnapshot()
  });

  it('should handleChange', () => {
    component = mount(
      <Dropdown 
        onChange={props.onChange} 
        label={label} 
        placeholder="- All Models -"  
        options={options} 
        selectedOption={options[0]} />
    );

    component.find('Select').prop('onChange')();

    expect(props.onChange).toHaveBeenCalled();
  })
})