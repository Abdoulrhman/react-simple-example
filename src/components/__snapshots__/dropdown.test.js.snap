// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component should render label correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dropdown
    label="models"
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "label": "Jeep",
          "value": "jeep",
        },
        Object {
          "label": "Tesla",
          "value": "tesla",
        },
      ]
    }
    placeholder="- All Models -"
    selectedOption={
      Object {
        "label": "Jeep",
        "value": "jeep",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          models
          :
        </label>,
        <StateManager
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          disabled={true}
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "Jeep",
                "value": "jeep",
              },
              Object {
                "label": "Tesla",
                "value": "tesla",
              },
            ]
          }
          placeholder="- All Models -"
          value={
            Object {
              "label": "Jeep",
              "value": "jeep",
            }
          }
        />,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "models",
            ":",
          ],
        },
        "ref": null,
        "rendered": Array [
          "models",
          ":",
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultInputValue": "",
          "defaultMenuIsOpen": false,
          "defaultValue": null,
          "disabled": true,
          "onChange": [Function],
          "options": Array [
            Object {
              "label": "Jeep",
              "value": "jeep",
            },
            Object {
              "label": "Tesla",
              "value": "tesla",
            },
          ],
          "placeholder": "- All Models -",
          "value": Object {
            "label": "Jeep",
            "value": "jeep",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            models
            :
          </label>,
          <StateManager
            defaultInputValue=""
            defaultMenuIsOpen={false}
            defaultValue={null}
            disabled={true}
            onChange={[Function]}
            options={
              Array [
                Object {
                  "label": "Jeep",
                  "value": "jeep",
                },
                Object {
                  "label": "Tesla",
                  "value": "tesla",
                },
              ]
            }
            placeholder="- All Models -"
            value={
              Object {
                "label": "Jeep",
                "value": "jeep",
              }
            }
          />,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "models",
              ":",
            ],
          },
          "ref": null,
          "rendered": Array [
            "models",
            ":",
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultInputValue": "",
            "defaultMenuIsOpen": false,
            "defaultValue": null,
            "disabled": true,
            "onChange": [Function],
            "options": Array [
              Object {
                "label": "Jeep",
                "value": "jeep",
              },
              Object {
                "label": "Tesla",
                "value": "tesla",
              },
            ],
            "placeholder": "- All Models -",
            "value": Object {
              "label": "Jeep",
              "value": "jeep",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Dropdown component should render label correctly 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dropdown
    label="models"
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "label": "Jeep",
          "value": "jeep",
        },
        Object {
          "label": "Tesla",
          "value": "tesla",
        },
      ]
    }
    placeholder="- All Models -"
    selectedOption={
      Object {
        "label": "Jeep",
        "value": "jeep",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          models
          :
        </label>,
        <StateManager
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          disabled={true}
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "Jeep",
                "value": "jeep",
              },
              Object {
                "label": "Tesla",
                "value": "tesla",
              },
            ]
          }
          placeholder="- All Models -"
          value={
            Object {
              "label": "Jeep",
              "value": "jeep",
            }
          }
        />,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "models",
            ":",
          ],
        },
        "ref": null,
        "rendered": Array [
          "models",
          ":",
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultInputValue": "",
          "defaultMenuIsOpen": false,
          "defaultValue": null,
          "disabled": true,
          "onChange": [Function],
          "options": Array [
            Object {
              "label": "Jeep",
              "value": "jeep",
            },
            Object {
              "label": "Tesla",
              "value": "tesla",
            },
          ],
          "placeholder": "- All Models -",
          "value": Object {
            "label": "Jeep",
            "value": "jeep",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            models
            :
          </label>,
          <StateManager
            defaultInputValue=""
            defaultMenuIsOpen={false}
            defaultValue={null}
            disabled={true}
            onChange={[Function]}
            options={
              Array [
                Object {
                  "label": "Jeep",
                  "value": "jeep",
                },
                Object {
                  "label": "Tesla",
                  "value": "tesla",
                },
              ]
            }
            placeholder="- All Models -"
            value={
              Object {
                "label": "Jeep",
                "value": "jeep",
              }
            }
          />,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "models",
              ":",
            ],
          },
          "ref": null,
          "rendered": Array [
            "models",
            ":",
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultInputValue": "",
            "defaultMenuIsOpen": false,
            "defaultValue": null,
            "disabled": true,
            "onChange": [Function],
            "options": Array [
              Object {
                "label": "Jeep",
                "value": "jeep",
              },
              Object {
                "label": "Tesla",
                "value": "tesla",
              },
            ],
            "placeholder": "- All Models -",
            "value": Object {
              "label": "Jeep",
              "value": "jeep",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Dropdown component should render options correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dropdown
    label="models"
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "label": "Jeep",
          "value": "jeep",
        },
        Object {
          "label": "Tesla",
          "value": "tesla",
        },
      ]
    }
    placeholder="- All Models -"
    selectedOption={
      Object {
        "label": "Jeep",
        "value": "jeep",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          models
          :
        </label>,
        <StateManager
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          disabled={true}
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "Jeep",
                "value": "jeep",
              },
              Object {
                "label": "Tesla",
                "value": "tesla",
              },
            ]
          }
          placeholder="- All Models -"
          value={
            Object {
              "label": "Jeep",
              "value": "jeep",
            }
          }
        />,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "models",
            ":",
          ],
        },
        "ref": null,
        "rendered": Array [
          "models",
          ":",
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultInputValue": "",
          "defaultMenuIsOpen": false,
          "defaultValue": null,
          "disabled": true,
          "onChange": [Function],
          "options": Array [
            Object {
              "label": "Jeep",
              "value": "jeep",
            },
            Object {
              "label": "Tesla",
              "value": "tesla",
            },
          ],
          "placeholder": "- All Models -",
          "value": Object {
            "label": "Jeep",
            "value": "jeep",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            models
            :
          </label>,
          <StateManager
            defaultInputValue=""
            defaultMenuIsOpen={false}
            defaultValue={null}
            disabled={true}
            onChange={[Function]}
            options={
              Array [
                Object {
                  "label": "Jeep",
                  "value": "jeep",
                },
                Object {
                  "label": "Tesla",
                  "value": "tesla",
                },
              ]
            }
            placeholder="- All Models -"
            value={
              Object {
                "label": "Jeep",
                "value": "jeep",
              }
            }
          />,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "models",
              ":",
            ],
          },
          "ref": null,
          "rendered": Array [
            "models",
            ":",
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultInputValue": "",
            "defaultMenuIsOpen": false,
            "defaultValue": null,
            "disabled": true,
            "onChange": [Function],
            "options": Array [
              Object {
                "label": "Jeep",
                "value": "jeep",
              },
              Object {
                "label": "Tesla",
                "value": "tesla",
              },
            ],
            "placeholder": "- All Models -",
            "value": Object {
              "label": "Jeep",
              "value": "jeep",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Dropdown component should render the right way 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dropdown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          :
        </label>,
        <StateManager
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          disabled={true}
          onChange={[Function]}
        />,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            ":",
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          ":",
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultInputValue": "",
          "defaultMenuIsOpen": false,
          "defaultValue": null,
          "disabled": true,
          "onChange": [Function],
          "options": undefined,
          "placeholder": undefined,
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            :
          </label>,
          <StateManager
            defaultInputValue=""
            defaultMenuIsOpen={false}
            defaultValue={null}
            disabled={true}
            onChange={[Function]}
          />,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              ":",
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            ":",
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultInputValue": "",
            "defaultMenuIsOpen": false,
            "defaultValue": null,
            "disabled": true,
            "onChange": [Function],
            "options": undefined,
            "placeholder": undefined,
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
